core/src/main/java/io/questdb/cutlass/text/types/SymbolAdapter.java

```diff
@@ -28,21 +28,29 @@
 import io.questdb.cairo.TableWriter;
 import io.questdb.cutlass.text.TextUtil;
 import io.questdb.std.str.DirectByteCharSequence;
-import io.questdb.std.str.DirectCharSink;
+import io.questdb.std.str.StringSink;
 
 public class SymbolAdapter extends AbstractTypeAdapter {
 
-    private final DirectCharSink utf8Sink;
+    private final StringSink utf8Sink;
+    private final boolean indexed;
 
-    public SymbolAdapter(DirectCharSink utf8Sink) {
-        this.utf8Sink = utf8Sink;
+    public SymbolAdapter(boolean indexed) {
+        this.indexed = indexed;
+        this.utf8Sink = new StringSink();
```

Comment:
I'd make `utf8Sink` a parameter in `write()` method instead of member variable and pass thread static into it from somewhere outside column / row loop.

Reply:
Thank you for that, it has been added :)